File: src/main/java/mods/betterfoliage/client/integration/GT5UIntegration.java
package mods.betterfoliage.client.integration;

import net.minecraft.block.Block;

import gregtech.common.pollution.ColorOverrideType;
import gregtech.common.pollution.Pollution;
import gregtech.common.pollution.PollutionConfig;

public final class GT5UIntegration {

    public static int applyPollutionTintStandard(int baseColor, Block block, int x, int z) {
        if (!Mod.GT5U.isLoaded() || !PollutionConfig.pollution) return baseColor;

        if (Pollution.standardBlocks == null) return baseColor;

        final ColorOverrideType colorOverride = Pollution.standardBlocks.matchesID(block);
        if (colorOverride == null) return baseColor;

        return colorOverride.getColor(baseColor, x, z);
    }
}

File: src/main/kotlin/mods/betterfoliage/client/integration/GT5UIntegration.kt
package mods.betterfoliage.client.integration

import cpw.mods.fml.relauncher.Side
import cpw.mods.fml.relauncher.SideOnly
import gregtech.common.pollution.Pollution
import gregtech.common.pollution.PollutionConfig
import net.minecraft.block.Block

@SideOnly(Side.CLIENT)
object GT5UIntegration {
    private const val STANDARD_TYPE = 0
    private const val CROSSED_SQUARES_TYPE = 1
    private const val LIQUID_TYPE = 4
    private const val VINE_TYPE = 20
    private const val DOUBLE_PLANT_TYPE = 40

    @JvmStatic fun tryTintWithPollution(originalColor: Int, block: Block, x: Int, z: Int): Int {
        if (!(Mod.GT5U.isLoaded && PollutionConfig.pollution)) return originalColor
        return when (block.renderType) {
            STANDARD_TYPE -> Pollution.standardBlocks
            CROSSED_SQUARES_TYPE -> Pollution.crossedSquares
            LIQUID_TYPE -> Pollution.liquidBlocks
            VINE_TYPE -> Pollution.blockVine
            DOUBLE_PLANT_TYPE -> Pollution.doublePlants
            else -> return originalColor
        }?.matchesID(block)?.getColor(originalColor, x, z) ?: originalColor
    }
}
